packages/playwright-core/src/cli/program.ts

```diff
@@ -563,9 +538,43 @@ async function launchContext(options: Options, extraOptions: LaunchOptions): Pro
     contextOptions.serviceWorkers = 'block';
   }
 
-  // Close app when the last window closes.
+  let browser: Browser;
+  let context: BrowserContext;
+
+  if (options.userDataDir) {
+    context = await browserType.launchPersistentContext(options.userDataDir, { ...launchOptions, ...contextOptions });
+    const persistentBrowser = context.browser();
+    assert(persistentBrowser, 'Could not launch persistent browser context');
```

Comment:
This assert looks strange, because it tests our public API in some random place 😄